source/matplot/backend/gnuplot.cpp

```diff
@@ -334,50 +349,24 @@ namespace matplot::backend {
 
     bool gnuplot::terminal_is_available(std::string_view term) {
         std::string msg = run_and_get_output("gnuplot -e \"set terminal " +
-                                     std::string(term.data()) + "\" 2>&1");
+                std::string{term} + "\" 2>&1");
         return msg.empty();
     }
 
-    std::tuple<int, int, int> gnuplot::gnuplot_version() {
-        static std::tuple<int, int, int> version{0, 0, 0};
-        const bool dont_know_gnuplot_version_yet =
-            version == std::tuple<int, int, int>({0, 0, 0});
-        if (dont_know_gnuplot_version_yet) {
-            std::string version_str =
-                run_and_get_output("gnuplot --version 2>&1");
-            std::string version_major = std::regex_replace(
-                version_str,
-                std::regex("[^]*gnuplot (\\d+)\\.\\d+ patchlevel \\d+ *"),
-                "$1");
-            std::string version_minor = std::regex_replace(
-                version_str,
-                std::regex("[^]*gnuplot \\d+\\.(\\d+) patchlevel \\d+ *"),
-                "$1");
-            std::string version_patch = std::regex_replace(
-                version_str,
-                std::regex("[^]*gnuplot \\d+\\.\\d+ patchlevel (\\d+) *"),
-                "$1");
-            try {
-                std::get<0>(version) = std::stoi(version_major);
-            } catch (...) {
-                std::get<0>(version) = 0;
-            }
-            try {
-                std::get<1>(version) = std::stoi(version_minor);
-            } catch (...) {
-                std::get<1>(version) = 0;
-            }
-            try {
-                std::get<2>(version) = std::stoi(version_patch);
-            } catch (...) {
-                std::get<2>(version) = 0;
-            }
-            const bool still_dont_know_gnuplot_version =
-                version == std::tuple<int, int, int>({0, 0, 0});
-            if (still_dont_know_gnuplot_version) {
-                // assume it's 5.2.6 by convention
-                version = std::tuple<int, int, int>({5, 2, 6});
+    Version gnuplot::gnuplot_version() {
+        static auto version = Version{};
+        if (!version) { // unknown version
+            const auto version_str = run_and_get_output("gnuplot --version 2>&1");
+            const auto major = word_after(version_str, "gnuplot");
+            const auto minor = word_after(major, ".");
```

Comment:
This pattern is a little confusing when the function is called repeatedly. Let's say the output is `"... gnuplot 5.2 patchlevel 6"`. Wouldn't `major` just be `"5"`? If so, then there's no word after `"."` because `major` doesn't include everything after `5`. That is, major is `"5"` and not `"5.2 patchlevel 6"`. On the other hand, if `major` does contain `"5.2 patchlevel 6"`, then the function name and return value are a little misleading.

Reply:
Right, the name `major` is misleading, I have renamed it to `major_minor` and added example in comments -- I think it's clearer when an example is upfront.